server/eclipse-project/src/main/java/us/freeandfair/corla/endpoint/ACVRDownload.java

```diff
@@ -87,23 +91,30 @@ public String endpoint(final Request the_request, final Response the_response) {
   }
   
   /**
-   * @return the set of cast vote records that were submitted by auditors.
+   * @return the set of cast vote records that were submitted by auditors, or 
+   * null if the query fails.
    */
   private Set<CastVoteRecord> getMatching() {
     Set<CastVoteRecord> result = null;
-
+    
     try {
-      final Set<CastVoteRecord> query_result = new HashSet<>();
       Persistence.beginTransaction();
       final Session s = Persistence.currentSession();
-      final Query<CastVoteRecord> query = 
-          s.createQuery("from CastVoteRecord where record_type = '" + 
-              RecordType.AUDITOR_ENTERED + "'", CastVoteRecord.class);
-      query_result.addAll(query.getResultList());
-      Persistence.commitTransaction();
-      result = query_result;
+      final CriteriaBuilder cb = s.getCriteriaBuilder();
+      final CriteriaQuery<CastVoteRecord> cq = 
+          cb.createQuery(CastVoteRecord.class);
+      final Root<CastVoteRecord> root = cq.from(CastVoteRecord.class);
+      cq.select(root).where(cb.equal(root.get("my_record_type"), 
```

Comment:
`my_record_type` is the field of CastVoteRecord that corresponds to the database column that we're comparing for equality. There really isn't a way to construct queries without knowing either database column names or field names for the entities being turned into database rows.